### ISSUE TITLE
States should be isolated

### ISSUE THREAD

**Issue by deadcoder0904** [vadimdemedes/ink-text-input#68]
I am trying to make a simple app with 2 inputs. One should run after another. But when I try to type, it types on both inputs even though I'm using `focus`.

# ui.tsx

```tsx
import React from 'react'
import { Box, Text } from 'ink'

import { state } from './state'
import { Input } from './Input'

const App = () => (
	<Box flexDirection="column">
		<Text color="#A78BFA">Compose New Post</Text>

		<Input
			label="Title"
			onSubmit={(title) => {
				state.title = title
			}}
		/>

		<Input
			label="Description"
			onSubmit={(description) => {
				state.description = description
			}}
		/>
	</Box>
)

module.exports = App
export default App
```

# Input.tsx

```tsx
import React from 'react'
import { Text } from 'ink'
import { UncontrolledTextInput } from 'ink-text-input'

interface IInput {
	label: string
	onSubmit: (s: string) => void
}

export const Input = ({ label, onSubmit }: IInput) => {
	const [submitted, setSubmitted] = React.useState(false)

	const handleSubmit = (q: string) => {
		onSubmit(q)
		setSubmitted(true)
	}

	if (submitted) return null

	return (
		<Text>
			<Text color="#6EE7B7">❯ {label}: </Text>
			<UncontrolledTextInput focus={!submitted} onSubmit={handleSubmit} />
		</Text>
	)
}
```

Checkout the main branch for a reproducable demo -> https://github.com/deadcoder0904/ink-isolated-state-bug

---

And the fix branch contains a potential fix -> https://github.com/deadcoder0904/ink-isolated-state-bug/tree/fix

### TL;DR of fix branch

---

It uses a `fieldNo` variable to move `focus` from 1 field to another.

# App.tsx

```tsx
import React from 'react'
import { Box, Text, useApp } from 'ink'

import { state } from './state'
import { Input } from './Input'

const App = () => {
	const [fieldNo, setFieldNo] = React.useState(1)
	const { exit } = useApp()

	React.useEffect(() => {
		if (fieldNo === 3) exit()
	}, [fieldNo])

	return (
		<Box flexDirection="column">
			<Text color="#A78BFA">Compose New Post</Text>

			<Input
				label="Title"
				focus={fieldNo === 1}
				onSubmit={(title) => {
					state.title = title
					setFieldNo(2)
				}}
			/>

			<Input
				focus={fieldNo === 2}
				label="Description"
				onSubmit={(description) => {
					state.description = description
					setFieldNo(3)
				}}
			/>
		</Box>
	)
}

module.exports = App
export default App
```

I don't think this should be needed. Is this how it's supposed to work? Because I'm unable to find any simple examples online using Multiple Text Inputs & I've searched for 2 days now.

Another thing is `exit` doesn't work here as it should. Am I doing anything wrong? Should I create a new issue?

**Comment by vadimdemedes** [comment]
There's a [`focus`](https://github.com/vadimdemedes/ink-text-input#focus) prop to enable or disable specific inputs, if you have more than one. It was there, but for some reason it was undocumented. Fixed it now.
